nvflare/apis/controller_spec.py

```diff
@@ -543,3 +543,20 @@ def cancel_all_tasks(self, completion_status=TaskCompletionStatus.CANCELLED, fl_
             fl_ctx: the FL context
         """
         pass
+
+    def wait_for_task(self, task: Task, abort_signal: Signal):
+        """Wait for task to complete or abort.
+
+        Args:
+            task: the task to wait for
+            abort_signal: the abort signal. If triggered, this method stops waiting and returns to the caller.
+        """
+        pass
+
+    def handle_dead_job(self, client_name: str, fl_ctx: FLContext):
+        """Called by the Engine to handle the case that the job on the client is dead.
+        Args:
+            client_name: name of the client on which the job is dead
+            fl_ctx: the FLContext
+        """
+        pass
```

Comment:
should these be inside the comm spec?

Reply:
`wait_for_task()` is only used in broadcast_operator.py as you mentioned above, maybe we no longer need this? If we can find an alternate approach for `handle_dead_job()` (such as with an event?), then I agree we can take this out of the controller_spec.

Reply:
We should revisit broadcast_operator.py. I'm not sure why it needs to implement multicasts_and_wait - the regular broadcast_and_wait should be able to do it.

Reply:
Removed `wait_for_task()` from controller. Also removed `handle_dead_job()` by adding an event.